### PR TITLE
feat: mui table fits workspace

### PR DESCRIPTION
**Pull request by Nezzely**
Partially addresses #74 

### Description of changes
- Updated the Criteria Table component to have tool bar
- Fixed table workspace to take up the entire area

### CODE REVIEW

File: web/src/modules/topic/components/CriteriaTable/CriteriaTable.tsx [hunk 1]
```diff
@@ -86,63 +83,63 @@ export const CriteriaTable = ({ problemNodeId }: Props) => {
...
-      </TableDiv>
-
       <Global styles={tableStyles} />
+      <MaterialReactTable
+        columns={columnData}
+        data={rowData}
+        enableColumnActions={false}
+        enablePagination={false}
+        enableBottomToolbar={false}
+        renderToolbarInternalActions={({ table }) => {
```
Comment (keyserj) [inline web/src/modules/topic/components/CriteriaTable/CriteriaTable.tsx:97]: I think it'll read a little nicer if we extract a `ToolbarActions` component (putting it just below the `columnData` definition seems fine) to pass to this prop, because right now you can't see all the of the table's props without scrolling

File: web/src/pages/index.page.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const Page = () => {
 
       <TopicToolbar />
 
-      <Box flex={1} position="relative">
+      <Box height="100%" position="relative">
```
Comment (keyserj) [inline web/src/pages/index.page.tsx:36]: np: I think it'd be good to use `const WorkspaceBox = styled(Box)` here to help establish the naming

File: web/src/modules/topic/components/CriteriaTable/CriteriaTable.tsx [hunk 3]
```diff
@@ -86,63 +83,63 @@ export const CriteriaTable = ({ problemNodeId }: Props) => {
...
+              <Tooltip title="Close">
+                <IconButton onClick={() => closeTable()} color="primary">
+                  <Cancel />
+                </IconButton>
+              </Tooltip>
+            </>
+          );
+        }}
+        enableSorting={false}
+        enableStickyHeader={true}
```
Comment (keyserj) [inline web/src/modules/topic/components/CriteriaTable/CriteriaTable.tsx:137]: pinning the first row could theoretically be in a separate commit but nbd

**Review by keyserj — COMMENTED** [review]
This is lookin STELLA'. Just some NPs. Also, if you manually link a PR to an issue, the issue will close when the PR merges https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue. I would probably just mention in the PR description something like `Partially addresses #74` so that there's a link between the PR and issue but the issue won't be closed.

File: web/src/pages/index.page.tsx (hunk 2 above)
Comment (Nezzely) [reply web/src/pages/index.page.tsx:36]: created a index.styles.tsx to hold custom component

**Review by keyserj — APPROVED** [review]
stella
